src/test/java/org/gridsuite/notification/server/NotificationWebSocketHandlerTest.java

```diff
@@ -123,15 +124,23 @@ private void withFilters(String filterStudyName, String filterUpdateType) {
         List<Map<String, Object>> expected = refMessages.stream().filter(headers -> {
             String name = (String) headers.get("studyName");
             String type = (String) headers.get("updateType");
+
             return (filterStudyName == null || filterStudyName.equals(name))
                     && (filterUpdateType == null || filterUpdateType.equals(type));
         }).collect(Collectors.toList());
         List<Map<String, Object>> actual = messages.stream().map(t -> {
             try {
-                var deserializedHeaders = ((Map<String, Map<String, Object>>) objectMapper.readValue(t, Map.class))
-                        .get("headers");
-                return Map.of("studyName", deserializedHeaders.get("studyName"), "updateType",
-                        deserializedHeaders.get("updateType"));
+                var deserializedHeaders = ((Map<String, Map<String, Object>>) objectMapper.readValue(t, Map.class)).get("headers");
+                var mapRes = new HashMap<String, Object>();
+
+                mapRes.put("studyName", deserializedHeaders.get("studyName"));
+                if (deserializedHeaders.get("updateType") != null) {
+                    mapRes.put("updateType", deserializedHeaders.get("updateType"));
+                }
+                if (deserializedHeaders.get("error") != null) {
+                    mapRes.put("error", deserializedHeaders.get("error"));
+                }
+                return mapRes;
             } catch (JsonProcessingException e) {
```

Comment:
Are these tests really necessary ?

Reply:
maybe not, but why putting a null value in the map ?

Reply:
See above